### ISSUE TITLE
Update Last User Interaction Timestamp more often

### ISSUE THREAD

**Issue by kriztan** [siacs/Conversations#1962]
#### General information
- **Version:** eg 1.13.6
- **Conversations source:** self build (latest HEAD)
#### Steps to reproduce
1. on device A switch display on time to some minutes, e.g. 10 minutes
2. on device A start Conversations on and open a chat, but don't close it
3. on device A wait some minutes e.g. 3 minutes, but let Conversations chat open and screen on 
4. on device B, the other participant could see a last seen time of 3 minutes for participant of device A
5. on device A write a message and let chat open
6. the participant on device B could see that user ob device A is typing
7. wait round about a minute and on device B, the other participant could see a last seen time of 4 minutes for participant of device A, olthough the user of device A wrote a message one minute ago
#### Expected result

The last seen timestamp should be updated when the user is typing, stopped typing, or scrolls the message list.
#### Debug output

There is no debug output.

**Comment by iNPUTmice** [comment]
> the other participant could see a last seen time of 4 minutes for participant of device 

active devices should have a timeout of 5 minutes in which they are considered 'online' (displayed as last seen just now

**Comment by kriztan** [comment]
What speaks against sending idle presences with last user input, too? So the timeout starts again.

**Comment by iNPUTmice** [comment]
Note to self: A solution would be to not time out the online status if the resource announces support for Last User interaction.

**Comment by castellanprime** [comment]
Hi, I want to solve this issue. Could you suggest ideas on how to begin?

**Comment by kriztan** [comment]
At the moment I'm testing a workaround and send presences without idle tags while user is typing a message to see if this is working. 
Take a look at https://github.com/kriztan/Pix-Art-Messenger/commit/1e4f06d04de38021a8cd62c386261103ebebccb6

**Comment by castellanprime** [comment]
kay.

**Comment by iNPUTmice** [comment]
@castellanprime if you want to do this particular teaser task then figuring out a sensible way to approach this is part of the task. The actual coding is probably not that much; the challenge is rather to get some understanding of XMPP.
We are not interested in hacks for this (sorry @kriztan :-) ) but a proper solution.

**Comment by castellanprime** [comment]
Tried to reproduce the intial issue with kritzan using [Genymotion](https://www.genymotion.com). The steps I took were:
(a) Create user on virtual device(Lets call that device Device B). My main phone is Device A.
(b) Create conference on Device A with the following settings: 

- Typing notifications (Enabled)
- Broadcast Last User Interaction (Enabled)
- Grace Period (Long)

(c) Wait three minutes.
(d) Start Coversation on Device B.
(e) In Device B, conference has a timestamp of just now.
(f) In Device A, send a message. 
(g) Message received in Device B updates its timestamp to match the one of device A.
(h) Wait some mins.  (In this interval, only device A updates its timestamp)
(i)  In Device B, type a long  message. (Device B' timestamp is updated as you type).
(j)  In Device B, send message. (updates its timestamp to match the one of device B).
(h) While all this is done, both devices do not show that user is typing.

In conclusion, I think that this work:
The last seen timestamp should be updated when the user is typing.

This does not work:
The last seen timestamp should be updated on both devices if user stopped typing, or scrolls the message list. 

Thanks.

**Comment by castellanprime** [comment]
@da2x I have a couple of questions about the flow of checking for the last seen timestamp. This is how I understand the flow:
(a) User will send an iq query packet to discover if the client server supports the 'idle' feature(xep-0319)
(b) The server responds with the list of features the server supports(I have added that to the Features flag in AbstractParser) 
(c) The PresenceParser checks for a presencepacket with the 'idle' feature. If it finds it, it sets the contact's last seen timestamp using the 'since' attribute while setting the contact as inactive. 
(d) The ContactDetailsActivity uses the Listeners to check for status changes. Basen on the contact's last seen, it uses the UI helper to set the human-readdable time for 'last seen' attribute
(e) The UI helper is where the 5 minutes from the xep-0319 specification is implemented. 
Is this the correct flow or am I missing something? Thanks.

**Comment by iNPUTmice** [comment]
@castellanprime 
b) the **client** responsds with the list of features

regarding d and e: The behavior right now is to show someone as idle when the idle tag is set (obviously) and to show someone as online when idle is not set AND when the last presence is less than 5 minutes old. (`!idle and (now() - last_presence <= 300)`).
Showing someone as idle can stay as it is; However the showing someone as online should be changed to: `!idle and hasFeature(idle-namespace)`
